pydy/tests/test_joints.py

```diff
@@ -0,0 +1,275 @@
+        assert cross(self.pinjoint.parent_axis, self.pinjoint.child_axis) == self.child.frame.z
+        assert cross(self.pinjoint.child_axis, self.pinjoint.parent_axis) == - self.parent.frame.z
+        self.align_axes()
+        parent_axis = self.pinjoint.parent_axis
+        child_axis = self.pinjoint.child_axis
+        # sample implementation
+        # perpendicular_axis_in_parent = - cross(parent_axis, child_axis)
+        # angle_between_axes = acos(dot(self.pinjoint.child_axis, self.pinjoint.parent_axis))
+        # self.child.frame.orient(self.parent.frame, 'Axis',
+        #                        [acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())),
+        #                         cross(parent_axis, child_axis)])
+        assert acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())) == 0
+
+        # part 4 adding angular velocity to child w.r.t parent.
+        assert self.child.frame.ang_vel_in(self.parent.frame) != 0
+
+    def test_pinjoint_apply_joint(self):
+        # apply_joint() should do everything done above by calling specific funtions
+        self.new_pinjoint = PinJoint('pinjoint', self.parent, self.child)
+        self.new_pinjoint._apply_joint()
+        # part 1
+        assert self.child.parent == self.parent
+        assert self.parent.child == self.child
+
+        # part 2
+        child_masscenter = self.child.masscenter
+        parent_masscenter = self.parent.masscenter
+        parent_point = parent_masscenter.locatenew(self.name + '_Point', self.parent_joint_vector)
+        child_point = child_masscenter.locatenew(self.name + '_Point', self.child_joint_vector)
+        assert self.pinjoint.parent_joint_vector == parent_point
+        assert self.pinjoint.child_joint_vector == child_point
+
+        # part 4
+        parent_axis = self.pinjoint.parent_axis
+        child_axis = self.pinjoint.child_axis
+        assert acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())) == 0
+
+
+class TestSlidingJoint():
+    def setup(self):
+        self.parent = Body('parent')
+        self.child = Body('child')
+        self.slidingjoint = SlidingJoint('slidingjoint', self.parent, self.child)
+
+    def test_slidingjoint_init(self):
+        # Default values
+        assert self.slidingjoint.parent == self.parent
+        assert self.slidingjoint.child == self.child
+        assert self.slidingjoint.direction1 == self.parent.x
+        assert self.slidingjoint.direction2 == self.child.x
+        assert self.slidingjoint.parent_point == self.parent.masscenter
+        assert self.slidingjoint.child_point == self.child.masscenter
+
+    def test_slidingjoint_paramters(self):
+        # custom parameters
+        self.slidingjoint = SlidingJoint('slidingjoint', self.parent, self.child, par_point_vec_tuple=(1,0,0),
+                                         child_point_vec_tuple=(0,1,0), direction1='x', direction2='y')
```

Comment:
You need to add a unit test for arbitrary directions.

Also please test using symbols with the `child_point_vec_tuple`.

I think `child_point` is better than `child_point_vec_tuple`. The docstring should explain that it is a three tuple.